### ISSUE TITLE
Improve DB state migration

### ISSUE THREAD

**Issue by pihme** [camunda/camunda#7248]
**Is your feature request related to a problem? Please describe.**
The current implementation of DB migration in `DbMigratorImpl` is triggered during the recovery of a broker. The migration run in a synchronous process.

This is not appropriate for migrations that take a long time.

**Describe the solution you'd like**
Two options have been discussed:
* Introduce a special mode in which migrations are run. This could be done as part of a rolling update process, if it exists. Or it could be a special start command for the Broker which runs the migration, then shuts down. This is the preferred long term solution.
* Make migrations utilize the actor scheduling options. In this scenario migrations, the normal stream processing should be paused until migrations are completed, and migrations should be implemented as asynchronous processes which can split their work into smaller chunks, where the chunks are then scheduled as actors.

**Comment by npepinpe** [comment]
Let's leave it in the back log for now, as the upcoming migrations are not expected to block too much. However, every time we introduce a migration we should consider if we need to implement this issue first.

**Comment by Zelldon** [comment]
Correct me if I'm wrong @npepinpe but isn't this issue about the data which is stored inside the ZeebeDB, which is engine related? So I would see this as part of the ZPA Team, since they are in charge of the data format of their data inside the db.

**Comment by npepinpe** [comment]
Yes, but I think the issue is specifically about the mechanism we use to run migration. If a migration is slow, it will take a while to run, and no processing is happening during that time. If you don't snapshot after running it, then you could potentially run it over and over. 

Someone could argue this falls more under infrastructure (like the stream processor) than engine specific. :shrug: 

I would also put it on the ZPA team for now, btw, but I can see where someone would draw the line differently

**Comment by marcosgvieira** [comment]
### 🧭 Triage Assessment

| Field | Value |
|------|-------|
| **Work type** | TASK_TOIL_NON_FEATURE |
| **Impact** | 🔥 High |
| **When** | ℹ️ Later |
| **Confidence** | 0.9 |

---

### 📝 Rationale

This is technical debt/operational work to change the broker migration mechanism and prevent long synchronous migrations from blocking processing, rather than a user-facing feature. The issue is sufficiently concrete and was explicitly left in the backlog because upcoming migrations were not expected to block significantly.  The work fits Core Features’ Zeebe engine and execution boundary, although ownership has previously been discussed as potentially overlapping with ZPA. 

---

🤖 _This issue was automatically triaged by an AI-based system as part of the Core Features triage workflow._

✋ **Want a human review?**
If you believe this assessment is incorrect or needs adjustment, feel free to update the details or request a manual triage by adding the **`triage:manual`** label.

**Comment by marcosgvieira** [comment]
### 🧭 Triage Assessment

| Field | Value |
|------|-------|
| **Work type** | TASK_TOIL_NON_FEATURE |
| **Impact** | 🔥 High |
| **When** | ℹ️ Later |
| **Confidence** | 0.9 |

---

### 📝 Rationale

Technical debt to change synchronous broker-recovery DB migrations that can block processing, rather than a user-facing capability. The issue is concrete, fits Core Features’ Zeebe engine/execution boundary, and was explicitly left for later because upcoming migrations were not expected to block significantly.   

---

🤖 _This issue was automatically triaged by an AI-based system as part of the Core Features triage workflow._

✋ **Want a human review?**
If you believe this assessment is incorrect or needs adjustment, feel free to update the details or request a manual triage by adding the **`triage:manual`** label.
